Sprint 7 solution in file manager

dendzim opened the pull request:
Добавлен менеджер сохраняющий задачи в файл.
Добавлен класс enum с типами задач.
Также добавлены тесты на новый менеджер.

dendzim committed:
Создана заглушка класса FileBackedTaskManager для фз 7.

dendzim committed:
Добавлена заглушка для класса тестирования

dendzim committed:
Добавлены методы в классы для enum Type

dendzim committed:
Реализован менеджер записывающий данные в файл

dendzim committed:
Добавлены тесты

dendzim committed:
Merge branch 'recovered-branch' into sprint_7-solution-in-file-manager

dendzim committed:
исправлен check style

dendzim committed:
исправлен check style

VadimZharkov commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,138 @@
+                if (task.getType() == TaskType.TASK) {
+                    taskManager.addTask(task);
+                } else if (task.getType() == TaskType.EPIC) {
+                    taskManager.addEpic((Epic) task);
```
Если так добавить, то в addEpic эпику присвоится новый id и тем самым разорвется связь между эпиком и его подзадачами. В addSubtask эпик просто не найдется. Нужно придумать как решить данную проблему.

VadimZharkov commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,138 @@
+                } else  {
+                    taskManager.addSubtask((Subtask) task);
+                }
+            }
```
Хорошо бы еще проинициализировать idCounter максимальным значением из файла, чтобы не было коллизий при последующем добавлении задач в менеджер.

VadimZharkov reviewed (changes requested):
Хорошая работа. Аккуратный стиль кода, так держать. Нужно исправить пару замечаний в части логики загрузки из файла.

dendzim committed:
Внесены правки согласно замечаниям.
Добавлен счетчик id инициализирующийся максимальным значением и исправлено добавление задач в коллекции.
